### PR TITLE
Update doc for ConservedVars dataclass

### PR DESCRIPTION
**Pull request by MTCam**
Start on and discuss better documentation for fluid data handling in *MIRGE-Com*.

### CODE REVIEW

**Comment by inducer** [comment]
It occurred to me that if `ConservedVars` supported just enough arithmetic, we could just use it as a state type and never even use the state-vector business. That might also help ferry temperature for restart (#139, #188). Possibly related: https://github.com/inducer/meshmode/pull/139.

**Comment by MTCam** [comment]
> It occurred to me that if `ConservedVars` supported just enough arithmetic, we could just use it as a state type and never even use the state-vector business. That might also help ferry temperature for restart (#139, #188). Possibly related: [inducer/meshmode#139](https://github.com/inducer/meshmode/pull/139).

In this case, it becomes something else, not a way to represent data that corresponds to the equations we are solving, so we'd need to rename it.

In general, I am not a giant fan of combining the two things - the conserved state, or quantities directly corresponding to the conserved state, vs. dependent quantities that we compute from the conserved state.  These are two separate things, and combining them on equal footing does not seem proper to me. (See e.g. PlasCom2)

Additionally, I think that having the dependent quantities ejected from the main data array has naturally forced some nice code constructions and make it clear *when and how* we compute things like dependent quantities.  If, on the other hand, the dependent quantities are embedded in the state array, it creates a situation where users are likely to do stuff like `project("vol", face, fv.temperature)`, for example, and this is super wrong.  Temperature, and many other dependent quantities *need* to be computed with the projected state, instead.   Having these quantities separated from the state forces you to be observant of where the data comes from.

File: mirgecom/fluid.py [hunk 1]
```diff
@@ -46,35 +46,172 @@
...
+
+        Often, a user starts with the fluid conserved quantities like mass and
+        energy densities, and it is desired to glom those quantities together into
+        a *MIRGE*-compatible $\mathbf{Q}$ data structure.
+
+        For example, a solution initialization routine may set the fluid
+        quantities::
+
+            den = rho  # rho is a DOFArray with fluid density
+            velo = v  # v is an obj array of ndim DOFArrays.
```
Comment (thomasgibson) [inline mirgecom/fluid.py:154]:
```suggestion
            velo = v  # v is an obj array of ndim DOFArrays denoting the components of velocity.
```

**Review by thomasgibson — COMMENTED** [review]
This is terrific @MTCam! Thank you for doing this. Just a super minor suggestion, but otherwise I'm quite happy with this! Tag me again once it's no longer a draft and I'm happy to give it my approval

File: mirgecom/fluid.py (hunk 1 above)
Comment (MTCam) [reply mirgecom/fluid.py:154]: I also wanted more text here, but ran into the line-length limit.  I will commit this and see if the linter screams at me.

Comment (MTCam) [reply mirgecom/fluid.py]: Yeah, it complained, and I could not figure out how to make the line continue inside the code block.   See how this kind-of-broken-english description grabs you (72fda48).

**Comment by MTCam** [comment]
> It occurred to me that if `ConservedVars` supported just enough arithmetic, we could just use it as a state type and never even use the state-vector business. That might also help ferry temperature for restart (#139, #188). Possibly related: [inducer/meshmode#139](https://github.com/inducer/meshmode/pull/139).

I'm re-thinking my previous knee-jerk response.  As you note, seems like if the ConservedVars (or StateThing) supported enough operations, then we'd never need to be flipping back-and-forth between "Q" and "CV", giving us a way to have a persistent "CV" object (or StateThing).

We had already been discussing, and toying around with memoizing some DV quantities onto the "StateThing" - and if i understand correctly, having a persistent StateThing would actually be **required** for that type of memoization/caching to be effective.

**Review by thomasgibson — APPROVED** [review]

File: mirgecom/fluid.py [hunk 2]
```diff
@@ -46,35 +46,172 @@
...
+        a *MIRGE*-compatible $\mathbf{Q}$ data structure.
+
+        For example, a solution initialization routine may set the fluid
+        quantities::
+
+            den = rho  # rho is a DOFArray with fluid density
+            velo = v  # v is an ndim-vector of DOFArray with components of velocity
+            ener = e  # e is a DOFArray with fluid energy
+            mom_dens = den*velo
+            ener_dens = den*ener
```
Comment (majosm) [inline mirgecom/fluid.py:157]:
```suggestion
             rho = ... # rho is a DOFArray with fluid density
             v = ...  # v is an ndim-vector of DOFArray with components of velocity
             e = ...  # e is a DOFArray with fluid energy
```

File: mirgecom/fluid.py [hunk 3]
```diff
@@ -46,35 +46,172 @@
...
+
+        An agglomerated array of fluid independent variables can then be
+        created with::
+
+            q = join_conserved(ndim, mass=den, energy=ener_dens, momentum=mom_dens)
```
Comment (majosm) [inline mirgecom/fluid.py:162]:
```suggestion
            q = join_conserved(ndim, mass=rho, energy=rho*e, momentum=rho*v)
```

**Review by majosm — COMMENTED** [review]
I'm too slow, but here are some minor formatting/wording suggestions.

**Pull request merged**